enterprise/frontend/src/metabase-enterprise/settings/index.ts

```diff
@@ -1,9 +1,29 @@
 import { hasAnySsoFeature } from "metabase/common/utils/plan";
+import { tokenFeatureObserver } from "metabase/embedding-sdk/utils/token-features-observer";
 import { isTest } from "metabase/env";
 import MetabaseSettings from "metabase/lib/settings";
 import type { TokenFeature } from "metabase-types/api";
 
+type Callback = () => void;
+export function hasPremiumFeatureAsync(
+  feature: TokenFeature,
+  callback: Callback,
+) {
+  const tokenFeatures = MetabaseSettings.get("token-features");
```

Comment:
Do we need this method to not be a promise/real "async"?
Otherwise we could simplify the code a lot by just exporting a promise "featureTokensLoaded", resolve it in `frontend/src/metabase/redux/settings.ts` and just doing

```
const tokens = await featureTokensLoaded
// rest of the logic
```